src/server/auth/server/api_server.go

```diff
@@ -713,6 +713,19 @@ func (a *apiServer) AddPipelineReaderToRepoInTransaction(txnCtx *txncontext.Tran
 	return a.setUserRoleBindingInTransaction(txnCtx, &auth.Resource{Type: auth.ResourceType_REPO, Name: sourceRepo}, auth.PipelinePrefix+pipeline, []string{auth.RepoReaderRole})
 }
 
+// AddPipelineReaderToRepoInTransaction gives a pipeline access to write data to the specified source repo.
+// The only time a pipeline needs write permission for a source repo is in the case of Cron inputs.
+// This is distinct from ModifyRoleBinding because AddPipelineWriter is a less expansive permission
+// that is included in the repoWriter role, versus being able to modify all role bindings which is
+// part of repoOwner. This method is for internal use and is not exposed as an RPC.
+func (a *apiServer) AddPipelineWriterToSourceRepoInTransaction(txnCtx *txncontext.TransactionContext, sourceRepo, pipeline string) error {
+	// Check that the user is allowed to add a pipeline to write to the output repo.
+	if err := a.CheckRepoIsAuthorizedInTransaction(txnCtx, &pfs.Repo{Type: pfs.UserRepoType, Name: sourceRepo}, auth.Permission_REPO_ADD_PIPELINE_WRITER); err != nil {
```

Comment:
Still thinking through the semantics here. Right now it's saying that a user needs write access to the source repo for a pipeline they created to write to it. I think this makes sense, but generally we think in terms of the pipelines output repo.

Reply:
Yeah I'm not sure what the ideal model is either, but this seems at least reasonable